src/signatures/secp256k1.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::convert::TryInto;
+use k256::{
+    elliptic_curve::{
+        ecdh,
+        generic_array::{typenum::Unsigned, GenericArray},
+        sec1::{Coordinates, EncodedPoint, ToEncodedPoint, UntaggedPointSize},
+    },
+    Secp256k1,
+};
+use signature::{Error as SigError, Signer, Verifier};
+
+/// A byte representation of an Secp256k1 field element.
+pub type FieldBytes = k256::FieldBytes;
+
+/// Secp256k1 Shared Secret - the result of a Diffie-Hellman key exchange.
+///
+/// Each party computes this from an (ephemeral) [SecretKey] and their counterparty's [PublicKey].
+pub type SharedSecret = k256::ecdh::SharedSecret;
+
+/// Secp256k1 Public Key
+pub struct PublicKey(k256::PublicKey);
+
+// algorithm identifier used for errors
+const NAME: &str = "Secp256k1";
+
+impl PublicKey {
+    /// Load a [PublicKey] from a slice of bytes.
+    pub fn from_bytes(bytes: &[u8]) -> crate::Result<Self> {
+        assert_buffer_eq!(bytes.len(), UntaggedPointSize::<Secp256k1>::USIZE, "bytes");
```

Comment:
Other SecretKey/PublicKey implementations in crypto.rs usually define SECRET_KEY_LEN/PUBLIC_KEY_LEN constants. Should it be the case here?

Reply:
Also, it'd would be best to stick to the same pattern when implementing wrappers for secret/public keys.